src/oid4vp/error.rs

```diff
@@ -0,0 +1,56 @@
+use super::request_signer::RequestSignerError;
+
+/// The [Error] enum represents the errors that can occur
+/// when using the oid4vp foreign library.
+#[derive(thiserror::Error, Debug, uniffi::Error)]
+pub enum Error {
+    #[error("An unexpected foreign callback error occurred: {0}")]
+    UnexpectedUniFFICallbackError(String),
+    #[error("Failed to validate the OID4VP Request: {0}")]
+    RequestValidation(String),
+    #[error("Failed to resolve the presentation definition: {0}")]
+    PresentationDefinitionResolution(String),
+    #[error("Failed to create verifiable presentation token: {0}")]
+    Token(String),
+    #[error("Unsupported Response Mode for OID4VP Request: {0}")]
+    UnsupportedResponseMode(String),
+    #[error("Failed to submit OID4VP response: {0}")]
+    ResponseSubmission(String),
+    #[error("Credential callback error: {0}")]
+    CredentialCallback(String),
+    #[error("Failed to create presentation submission: {0}")]
+    PresentationSubmissionCreation(String),
+    #[error("Failed to parse DID url: {0}")]
+    InvalidDIDUrl(String),
+    #[error("Failed to generate DID key URL: {0}")]
+    DIDKeyGenerateUrl(String),
+    #[error("Failed to encode credential: {0}")]
+    CredentialEncodingError(String),
+    #[error("Failed to decode credential: {0}")]
+    CredentialDecodingError(String),
+    #[error("Failed to parse JSON syntax: {0}")]
+    JsonSyntaxParse(String),
+    #[error(transparent)]
+    VdcCollection(#[from] crate::vdc_collection::VdcCollectionError),
```

Comment:
Interestingly, Kotlin seemed to have no issues building with this `#[from]` macro.

TODO: re-check if other error types can be re-added back for the conversation.